src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -583,10 +585,11 @@ def _remove_nulls(managed_clusters):
         for attr in attrs:
             if getattr(managed_cluster, attr, None) is None:
                 delattr(managed_cluster, attr)
-        for ap_profile in managed_cluster.agent_pool_profiles:
-            for attr in ap_attrs:
-                if getattr(ap_profile, attr, None) is None:
-                    delattr(ap_profile, attr)
+        if managed_cluster.agent_pool_profiles is not None:
```

Comment:
This workaround is not longer required. AgentPools will not be null ?

Reply:
that is correct. Will remove this.